### PR TITLE
Adds Automated Testing & Code Quality Tools

### LINKED ISSUE #2: Add test for parsing microservice

### PR DESCRIPTION
**Pull request by hkeeler**
Closes #2.

This is a first take at adding automated testing and code quality tools to the grasshopper-parser API.  These tests should be considered semi-integration tests in that they focus on testing the REST API, and less so on any underlying logic, such as what goes on under the hood with the `validate` param.

These tests are written using the [Flask Testing](http://flask.pocoo.org/docs/0.10/testing/) setup, which really uses a [Werkzeug Test Client](http://werkzeug.pocoo.org/docs/0.10/test/#werkzeug.test.Client) under the hood for the heavy lifting.  In addition, I am using [nose](https://nose.readthedocs.org/en/latest/) as the unittesting framework, [coverage](http://nedbatchelder.com/code/coverage/) for test coverage, [flake8](http://flake8.readthedocs.org/en/latest/index.html) for code quality, and [tox](https://tox.readthedocs.org/en/latest/) to glue it all together.

### CODE REVIEW

**Comment by hkeeler** [comment]
https://travis-ci.org appears to be "Offline for Maintenance" at the moment.  Please hold off merging until I can confirm the that Travis setup is actually working.

**Comment by hkeeler** [comment]
OK, Travis seems to be happy, _and_ I've fixed the issue causing flake8 to blow up.  It wasn't _really_ blowing up, its just that it defaults to a `1` exit code when it breaks one of its code quality rules.  This was especially confusing as it returns the following, making you think something is broken in the python path:

``` python
ERROR: InvocationError: '/home/travis/build/cfpb/grasshopper-parser/.tox/flake8/bin/flake8'
```

I _personally_ don't really consider this to be a fail.  To fix this, I added the `--exit-zero` arg to the `flake8` call.  All is well now, and ready for full review.

**Comment by hkeeler** [comment]
I've also added Coveralls (https://coveralls.io/r/hkeeler/grasshopper-parser) into this PR.

Closes #8.

File: app.py [hunk 1]
```diff
@@ -52,39 +52,40 @@ def parse_with_tag(addr_str):
...
-parse_method_dispatch = {'parse' : parse_with_parse, 'tag' : parse_with_tag}
+parse_method_dispatch = {'parse': parse_with_parse, 'tag': parse_with_tag}
 
 
 class InvalidApiUsage(Exception):
     """
     Exception for invalid usage of address parsing API
 
-    This is a simplifiled version of 
+    This is a simplifiled Flask's Implementing API Exceptions:
```
Comment (wpears) [inline app.py]: Probably meant  **version of** Flask's

**Comment by wpears** [comment]
LGTM. I'll merge after the typo fix above.

**Comment by hkeeler** [comment]
Cool.  Should be all good now.  Thanks @wpears.

**Pull request merged**